### PR TITLE
Fix httpasyncclient-4.x-plugin's memory leak risk

### PR DESCRIPTION
**Pull request by PepoRobert**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
It is hard to check as it is contributed from global people. Please provide analysis such as process flow and heap dump to describe the leak.
I am limited to determine this from reading source codes.

**Comment by wu-sheng** [comment]
@heihaozi Could you confirm this? I think this interceptor was introduced by you through https://github.com/apache/skywalking-java/pull/40

**Comment by PepoRobert** [comment]
The next interceptor thread has switched&nbsp; remove&nbsp;not take effect






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/skywalking-java"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月17日(星期五) 晚上7:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [apache/skywalking-java] threadlocal has memory leak risk   should be removed after use (PR #85)





 
@heihaozi Could you confirm this? I think this interceptor was introduced by you through #40
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by wu-sheng** [comment]
OK, we are waiting for the original author to verify. @PepoRobert  You need to update the `changes.md` in the update to describe this fix.

File: changes/changes-8.8.0.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ Release Notes.
 * Add plugin to support okhttp 2.x
 * Optimize okhttp 3.x 4.x plugin to get span time cost precisely
 * Adapt message header properties of RocketMQ 4.9.x
+* fix the bug httpasyncclient-4.x-plugin  has memory leak risk
```
Comment (wu-sheng) [inline changes/changes-8.8.0.md:43]:
```suggestion
* Fix httpasyncclient-4.x-plugin's memory leak risk
```

**Review by wu-sheng — APPROVED** [review]
Good to me. Will wait for another 2 days, in case @heihaozi has things to comment.

**Comment by heihaozi** [comment]
Good to me.

**Pull request merged**